src/zenml/stack_stores/base_stack_store.py

```diff
@@ -304,7 +328,55 @@ def __check_component(
             typ: name for typ, name in map(__check_component, stack.components)
         }
         metadata = {c.type.value: c.flavor for c in stack.components}
-        self._create_stack(stack.name, stack_configuration)
+        self._save_stack(stack.name, stack_configuration)
+        logger.info("Registered stack with name '%s'.", stack.name)
+        return metadata
+
+    def update_stack(self, name: str, stack: StackWrapper) -> Dict[str, str]:
+        """Update a stack and its components.
+
+        If any of the stack's components aren't registered in the stack store
+        yet, this method will try to register them as well.
+
+        Args:
+            name: The original name of the stack.
+            stack: The new stack to use in the update.
+
+        Returns:
+            metadata dict for telemetry or logging.
+
+        Raises:
+            DoesNotExistException: If no stack exists with the given name.
+        """
+        try:
+            self.get_stack(name)
+        except KeyError:
+            raise DoesNotExistException(
+                f"Unable to update stack with name '{stack.name}': No existing "
+                f"stack found with this name."
+            )
+
+        def __check_component(
+            component: StackComponentWrapper,
+        ) -> Tuple[StackComponentType, str]:
+            try:
+                _ = self.get_stack_component(
+                    component_type=component.type, name=component.name
+                )
+            except KeyError:
+                self.register_stack_component(component)
+            return component.type, component.name
+
```

Comment:
Same as the stack component, I think we should still have a check here if something with the new name already exists. Probably a StackExistsError?